### PR TITLE
Refactor hipBLASLt documentation

### PR DESCRIPTION
**Pull request by SwRaw**
Refactoring all libraries to adhere to the documentation standards that includes following home page format, index style, language improvements, link fixes, navigation, and overall content readability improvements.

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+Prerequisites
+=============
+
+* A ROCm enabled platform, for more information refer to `ROCm Documentation <https://rocm.docs.amd.com/>`_.
+* A compatible version of hipBLAS
+
+Installing pre-built packages
+=============================
+
+Download pre-built packages either from `ROCm's package servers <https://rocm.github.io/install.html#installing-from-amd-rocm-repositories>`_ or by clicking the GitHub releases tab and manually downloading, which could be newer.  Release notes are available for each release on the releases tab.
```
Comment (SwRaw) [inline docs/installation.rst]: The link <https://rocm.github.io/install.html#installing-from-amd-rocm-repositories>` is not correct. Please refer rocm.docs.amd.com and provide the correct link. Also, please review the entire paragraph on installing prebuilt packages and ensure that it is still valid.

Comment (yhuiYH) [reply docs/installation.rst]: @SwRaw can you revise the paragraph, based on what was done in hipBLAS: https://github.com/ROCm/hipBLAS/blob/develop/docs/install.rst

Comment (SwRaw) [reply docs/installation.rst]: @yhuiYH What are we asking them to download from here: https://github.com/ROCm/hipBLAS/releases? This page has release notes.

Comment (jichangjichang) [reply docs/installation.rst]: @SwRaw  It should be https://github.com/ROCm/hipBLASLt/releases but not https://github.com/ROCm/hipBLAS/releases

Comment (SwRaw) [reply docs/installation.rst]: @jichangjichang Ok. But it still has release notes only. We are asking users to manually download packages from this page.

**Review by jichangjichang — APPROVED** [review]
lgtm

**Comment by jichangjichang** [comment]
@SwRaw  Please let me know once you would like to merge it.

File: docs/installation.rst (hunk 1 above)
Comment (SwRaw) [reply docs/installation.rst]: @jichangjichang Can we address this please: "https://github.com/ROCm/hipBLAS/releases has no packages to be downloaded. Should I remove the line where we ask them to download?" 

Ref:
Installing prebuilt packages
Download prebuilt packages either from `ROCm's native package manager <https://rocm.docs.amd.com/projects/install-on-linux/en/latest/tutorial/quick-start.html#native-package-manager>`_ or by going to the `GitHub releases <https://github.com/ROCm/hipBLAS/releases>`_ and manually downloading, which could be newer.  Release notes are available for each release on the releases tab.

Comment (jichangjichang) [reply docs/installation.rst]: @SwRaw   I would recommend to remove GitHub release link.

Comment (SwRaw) [reply docs/installation.rst]: Done!
@lpaoletti Please review.

**Comment by lpaoletti** [comment]
**installation.rst**
This is messed up:
image

This seems wrong - two headings in a row without body text in between:
image


image

**Comment by SwRaw** [comment]
@lpaoletti Thanks for your review. The code blocks were not being rendered, that's why stacked headings. I should have noticed it earlier.  Fixed it and reviewed the doc again. Please take a quick look again and let me know if the PR is good to be merged.

**Review by lpaoletti — APPROVED** [review]

**Pull request merged**